openshift_tools/cloud/aws/ebs_snapshotter.py

```diff
@@ -324,42 +330,55 @@ def get_expired_snapshots(snapshots, hourly_backups, daily_backups, weekly_backu
         # We want to make sure we're only sending back 1 copy of each snapshot to trim.
         return list(set(snaps_to_trim))
 
-    @staticmethod
-    def get_volume_snapshots(volume, all_snapshots):
-        """ Filters out the snapshots for a given volume from the entire list of snapshots. """
-        # Get this volume's specific snapshots
-        vol_snaps = [s for s in all_snapshots if s.volume_id == volume.id]
-        EbsSnapshotter.sort_snapshots(vol_snaps)
-        return vol_snaps
 
     # Reason: disable pylint too-many-arguments because this is the API I want to expose.
     # Status: permanently disabled
     # pylint: disable=too-many-arguments
-    def trim_snapshots(self, hourly_backups, daily_backups, weekly_backups, monthly_backups, dry_run=False):
+    # pylint: disable=too-many-statements
+    def trim_snapshots(self, hourly_backups,
+                       daily_backups,
+                       weekly_backups,
+                       monthly_backups,
+                       delete_orphans_older_than,
+                       dry_run=False):
```

Comment:
This method is now very big. Consider spending some time breaking out some of its functionality into separate methods.